desc/objectives/_coils.py

```diff
@@ -854,13 +854,28 @@ def build(self, use_jit=True, verbose=1):
             Bplasma = compute_B_plasma(
                 eq, eval_grid, self._source_grid, normal_only=True
             )
+        field = self._field
+        if hasattr(field, "Phi_mn"):
+            # make the transform for the FourierCurrentPotentialField
+            if self._field_grid is None:
+                self._field_grid = LinearGrid(
+                    M=30 + 2 * max(field.M, field.M_Phi),
+                    N=30 + 2 * max(field.N, field.N_Phi),
+                    NFP=field.NFP,
+                )
+            field_transforms = get_transforms(
+                ["K", "x"], obj=field, grid=self._field_grid
+            )
+        else:
+            field_transforms = None
```

Comment:
I don't think I understand this. Why is `field_transforms = None` when the field grid is `None`?

Reply:
Other field types (well other then CurrentPotentialField) don't need these pre-computed specially (well I guess a curve could but those are fine being created on the fly)